src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -15,7 +14,7 @@ namespace Microsoft.Extensions.DependencyInjection
     /// <summary>
     /// The default IServiceProvider.
     /// </summary>
-    internal class ServiceProvider : IServiceProvider, IDisposable
+    public sealed class ServiceProvider : IServiceProvider, IDisposable
```

Comment:
Aren't any virtual calls; might as well make it non-sealed? (Then people can extend/enhance in a non-interfering way)